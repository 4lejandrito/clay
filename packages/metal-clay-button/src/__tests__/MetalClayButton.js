'use strict';

import MetalClayButton from '../MetalClayButton';

let button;

describe('MetalClayButton', function() {
	afterEach(() => {
		if (button) {
			button.dispose();
		}
	});

	it('should generate the default markup', function() {
		button = new MetalClayButton();

		expect(button).toMatchSnapshot();
	});

	it('should render a submit button', function() {
		button = new MetalClayButton({
			type: 'submit',
		});

		expect(button).toMatchSnapshot();
	});

	it('should render a disabled button', function() {
		button = new MetalClayButton({
			disabled: true,
		});

		expect(button).toMatchSnapshot();
	});

	it('should render a success button', function() {
		button = new MetalClayButton({
			style: 'success',
		});

		expect(button).toMatchSnapshot();
	});

	it('should render a button with label', function() {
		button = new MetalClayButton({
			label: 'Label',
		});

		expect(button).toMatchSnapshot();
	});

	it('should render a button with label', function() {
		button = new MetalClayButton({
			name: 'myButton',
		});

		expect(button).toMatchSnapshot();
	});

	it('should render a link with ariaLabel attribute', function() {
		button = new MetalClayButton({
			ariaLabel: 'My Description',
			href: 'http://liferay.com',
			label: 'Visit Liferay.com',
		});

		expect(button).toMatchSnapshot();
	});
});
