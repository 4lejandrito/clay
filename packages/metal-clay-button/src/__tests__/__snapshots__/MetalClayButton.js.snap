// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetalClayButton should generate the default markup 1`] = `<button class="btn btn-primary"></button>`;

exports[`MetalClayButton should render a button with label 1`] = `<button class="btn btn-primary">Label</button>`;

exports[`MetalClayButton should render a button with label 2`] = `<button class="btn btn-primary" name="myButton"></button>`;

exports[`MetalClayButton should render a disabled button 1`] = `<button class="btn btn-primary" disabled=""></button>`;

exports[`MetalClayButton should render a link with ariaLabel attribute 1`] = `<a href="http://liferay.com" class="btn btn-primary" aria-label="My Description">Visit Liferay.com</a>`;

exports[`MetalClayButton should render a submit button 1`] = `<button class="btn btn-primary" type="submit"></button>`;

exports[`MetalClayButton should render a success button 1`] = `<button class="btn btn-success"></button>`;
